docs/StandardLibrary.md

```diff
@@ -0,0 +1,334 @@
+# Flux Standard Library Design
+
+The Flux standard library will define the set of functions, constants, and other
+values that are available out of the box with Flux, providing the functionality that is
+most frequently used.
+
+Some functions in the standard library will be available without qualifying
+them with a namespace.  These will be the most common functions, to help Flux
+code be more concise.  Such functions are said to be in the "top level package."
+
+The remaining functions in the standard library will be divided up into
+discrete packages that are organized by topic.
+
+What's presented here is just a departure point for further discussion.
+
+## The Top Level Package
+
+Those functions that provide functionality that is very frequently
+used and should not need to be qualified with a namespace in Flux code.
+(Consider also the concept of "prelude")
+
+### Transformations
+Should some of these be in the `math` package instead?  `pearsonr` for example.
+
+Fundamental transformations:
+- `bottom`
+- `distinct`
+- `filter`
+- `group`
+- `join`
+- `keyValues`
+- `keys`
+- `limit`
+- `map`
+- `pivot`
+- `range`
+- `set`
+- `shift`
+- `sort`
+- `stateCount`
+- `stateDuration`
+- `top`
+- `union`
+- `window`
+- `yield`
+
+
+Schema changing operations:
+- `drop`
+- `duplicate`
+- `keep`
+- `rename`
+
+Aggregate operations:
+- `aggregateWindow`
+- `covariance`
+- `cov`
+- `pearsonr`
+- `count`
+- `integral`
+- `mean`
+- `median`
+- `percentile`
+- `skew`
+- `spread`
+- `stddev`
+- `sum`
+
+Selector operations:
+- `first`
+- `last`
+- `max`
+- `min`
+- `percentile`
+- `median`
+- `sample`
+
+Highest/Lowest
+- `highestMax`
+- `highestAverage`
+- `highestCurrent`
+- `lowestMax`
+- `lowestAverage`
+- `lowestCurrent`
+
+Binning transformations:
+- `histogram`
+- `histogramQuantile`
+- `linearBins`
+- `logarithmicBins`
+
+Other mathematical operations:
+- `cumulativeSum`
+- `derivative`
+- `difference`
+
+## Package `testing`
+- `assertEquals`
+
+Others?
+
+## I/O Packages
+
+Should vendor-specific packages like `influx` be part of the standard library?  
+If Flux is to be used as a stand-alone language independent of InfluxDB, perhaps
+they should not.
+
+### Package `csv`
+- `from` (seems lonely here; could be in top level package?)
```

Comment:
I think this is fine. Also I expect we could have some kind of `io.from` that takes a raw byte stream and produces a stream of tables. Then the `csv` package will have a `encoder` and `decoder` types.

And `csv.from` would be defined in terms of the `io` package:

    package csv
    from = (file,content) => io.from(encoder: decoder(file:file, contents:contents))

Or something similar. Obviously the Flux type system doesn't support this yet but we can figure that out separately.

In short I expect the csv package to grow.

Reply:
I like it.  Curious about what the package structure would look like if we went this route.  Would we need a set of transport packages, each of which provide a `from` and `to` that require encoders or decoders as arguments?
```
package http
package file
package kafka
...
```
As well as a set of encoder/decoder packages?  They could also provide `from` and `to` functions that use a default transport for the encoding.
```
package csv
package influx // both LP and Influx JSON
...
```
(Unclear to me where `sql` would fit in this kind of abstraction)